orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/model/V2PipelineTemplate.java

```diff
@@ -0,0 +1,307 @@
+      if (overlay.type != null) { v.type = overlay.type; }
+      if (overlay.defaultValue != null) { v.defaultValue = overlay.defaultValue; }
+      if (overlay.example != null) { v.example = overlay.example; }
+      return v;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+      return super.clone();
+    }
+  }
+
+  public static class Configuration {
+    private Map<String, Object> concurrentExecutions;
+    private List<NamedHashMap> triggers;
+    private List<NamedHashMap> parameters;
+    private List<NamedHashMap> notifications;
+    private List<HashMap<String, Object>> expectedArtifacts;
+
+    public Map<String, Object> getConcurrentExecutions() {
+      return Optional.ofNullable(concurrentExecutions).orElse(new HashMap<>());
+    }
+
+    public void setConcurrentExecutions(Map<String, Object> concurrentExecutions) {
+      this.concurrentExecutions = concurrentExecutions;
+    }
+
+    public List<NamedHashMap> getTriggers() {
+      return triggers;
+    }
+
+    public void setTriggers(List<NamedHashMap> triggers) {
+      this.triggers = triggers;
+    }
+
+    public List<NamedHashMap> getParameters() {
+      return parameters;
+    }
+
+    public void setParameters(List<NamedHashMap> parameters) {
+      this.parameters = parameters;
+    }
+
+    public List<NamedHashMap> getNotifications() {
+      return notifications;
+    }
+
+    public void setNotifications(List<NamedHashMap> notifications) {
+      this.notifications = notifications;
+    }
+
+    public List<HashMap<String, Object>> getExpectedArtifacts() {
+      return expectedArtifacts;
+    }
+
+    public void setExpectedArtifacts(List<HashMap<String, Object>> expectedArtifacts) {
+      this.expectedArtifacts = expectedArtifacts;
+    }
+  }
+
+  @Override
+  @JsonIgnore
+  public String getSchemaVersion() {
+    return schema;
+  }
+
+  public String getSchema() {
+    return schema;
+  }
+
+  public void setSchema(String schema) {
+    this.schema = schema;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public void setId(String id) {
+    this.id = id;
+  }
+
+  public Metadata getMetadata() {
+    return metadata;
+  }
+
+  public void setMetadata(Metadata metadata) {
+    this.metadata = metadata;
+  }
+
+  public Boolean getProtect() {
+    return protect;
+  }
+
+  public void setProtect(Boolean protect) {
+    this.protect = protect;
+  }
+
+  public List<Variable> getVariables() {
+    return variables;
+  }
+
+  public void setVariables(List<Variable> variables) {
+    this.variables = variables;
+  }
+
+  public Configuration getConfiguration() {
+    if (configuration == null) {
+      configuration = new Configuration();
+    }
+    return configuration;
+  }
+
+  public void setConfiguration(Configuration configuration) {
+    this.configuration = configuration;
+  }
+
+  public List<StageDefinition> getStages() {
+    return Optional.ofNullable(stages).orElse(Collections.emptyList());
```

Comment:
You use this optional pattern here, but 10 lines up you explicitly check for null when returning configuration.

Reply:
Done.